telemetry/wrapper.go

```diff
@@ -19,6 +20,18 @@ func NewLabel(name, value string) metrics.Label {
 	return metrics.Label{Name: name, Value: value}
 }
 
+// ModuleMeasureSinceWithSampling samples latency metrics given the sample rate.
+// This is intended to be used in hot code paths.
+func ModuleMeasureSinceWithSampling(module string, start time.Time, sampleRate float64, keys ...string) {
+	if rand.Float64() < sampleRate {
```

Comment:
There are a lot of issues with sampling based upon a probabilistic rate since:
* it usually requires a lot of tuning over time.
* you can't be sure if the telemetry information is skewed due to low frequency of events or bad random number generator sequence.

Sampling per N units of time makes a lot more sense. For example only allow this metric to be sampled once every second. It would also be good to scale the sample based upon how many times it has occurred correctly but that seems like it would require changing the `go-metrics` package to take `bulk` updates for samples.

If you still want to go down this path then it would make sense to use a faster random number generator like https://github.com/flyingmutant/rand and also ensure that you perform the sampling before you compute the time which would require the sampling to happen in the callers method (which is annoying without macros).

Reply:
As discussed on slack, we will go as is.

We should swap to use float32 as we don't need the extra precision that float64 provides and it takes less effort to RNG 32 bits.

Reply:
it actually takes more effort because Float32 internally calls Float64 😅

Reply:
That is terrible. Likely why other `rand` libraries are so common in `golang`.